mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt

```diff
@@ -650,12 +650,14 @@ class PreferencesActivity : AbstractBaseActivity() {
         ConfirmationDialogFragment.Callback,
         ServerEditorConfirmLeaveDialogFragment.Callback {
         private lateinit var config: ServerConfiguration
+        private lateinit var initialConfig: ServerConfiguration
         private var markAsPrimary = false
 
         override val titleResId: Int get() = R.string.settings_edit_server
 
         override fun onCreate(savedInstanceState: Bundle?) {
             config = requireArguments().getParcelable("config")!!
+            initialConfig = config
```

Comment:
Dies this actually work? I'd expect only the reference to be copied and thus later modifications to `config` also being applied to `initialConfig`.

Reply:
It copies the reference here, but when `config` is changed, a new object is created and only assigned to `config`, not `initialConfig`.

Reply:
Ah, right, due to `ServerConfiguration` being immutable. Nevermind then :-)